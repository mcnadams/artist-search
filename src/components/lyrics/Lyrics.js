import React from 'react';
import PropTypes from 'prop-types';

function Lyrics({ lyrics }) {
  const lyricsP = lyrics.split('\n');
  const lyricsParas = lyricsP.map((paragraph, i) => {
    return (
      <p key={i}>
        {paragraph}
      </p>
    );
  });  
  return (
    <div>
      {lyricsParas}
    </div>
  );
}

Lyrics.propTypes = {
  lyrics: PropTypes.string.isRequired
};

export default Lyrics;
