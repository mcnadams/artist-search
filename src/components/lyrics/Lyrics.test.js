import React from 'react';
import { shallow } from 'enzyme';
import Lyrics from './Lyrics';

describe('Lyrics renders', () => {
  it('renders Lyrics', () => {
    const wrapper = shallow(<Lyrics lyrics={'lady'} />);
    expect(wrapper).toMatchSnapshot();
  });
});
