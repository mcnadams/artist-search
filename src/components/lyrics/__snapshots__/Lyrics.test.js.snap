// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lyrics renders renders Lyrics 1`] = `
<div
  style={
    Object {
      "background": "white",
      "textAlign": "center",
    }
  }
>
  <p
    key="0"
  >
    lady
  </p>
</div>
`;
